ui/src/permission.js

```diff
@@ -60,21 +60,29 @@ router.beforeEach((to, from, next) => {
       next({ path: '/dashboard' })
       NProgress.done()
     } else if (to.path === '/verify2FA' || to.path === '/setup2FA') {
-      if (store.getters.twoFaEnabled && !store.getters.loginFlag) {
+      const isSAML = JSON.parse(Cookies.get('isSAML') || Cookies.get('isSAML', { path: '/client' }) || false)
```

Comment:
@harikrishna-patnala suggestion: we shouldn't worry if this is 2FA enabled or not; a general way to implement is to simply test if 2FA is required and simply redirect to the appropriate path for the logged in user. (we assume once SAML SSO has worked, the user has logged in)

Reply:
yeah @rohityadavcloud but we needed more details if 2FA is enabled or 2FA is configured to redirect to appropriate page. so for that I kept these conditions.